Algebra/PartialOrd.hs

```diff
@@ -39,11 +39,14 @@ import qualified Data.IntMap as IM
 --
 -- @
 -- Reflexive:  a == a
--- Transitive: a == b && b == c ==> a == b
+-- Transitive: a == b && b == c ==> a == c
 -- @
 class Eq a => PartialOrd a where
     leq :: a -> a -> Bool
 
+    comparable :: a -> a -> Bool
```

Comment:
I'd like to have a haddock explaining what *comparable* is (algebraically), and maybe some examples (positive and negative).

Also an implementation for `Ordered` would be good to have, as it's trivially `comparable _ _ = True`.